File: WebApplication/src/suppliers/dtos/search/supplier-group/base/base-supplier-group.dto.ts
import { SmartSpecializationGroupDto } from "../../supplier-offering-group/smart-specialization-group.dto";

export class BaseSupplierGroupDto {
    id: number;

    // Institution
    uic: string;
    code: string;

    // Common
    name: string;
    nameAlt: string;
    viewOrder: number

    smartSpecializations: SmartSpecializationGroupDto[] = [];
}

File: Sc.Models/Dtos/Suppliers/Search/SupplierGroup/Base/BaseSupplierGroupDto.cs
using Sc.Models.Dtos.Suppliers.Search.SupplierOfferingGroup;

namespace Sc.Models.Dtos.Suppliers.Search.SupplierGroup.Base
{
    public abstract class BaseSupplierGroupDto
    {
        public int? Id { get; set; }

        // Institution
        public string Uic { get; set; }
        public string Code { get; set; }

        // Common
        public string Name { get; set; }
        public string NameAlt { get; set; }
        public int ViewOrder { get; set; }

        public List<SmartSpecializationGroupDto> SmartSpecializations { get; set; } = new List<SmartSpecializationGroupDto>();
    }
}
